lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/ports/ListTcpPortProvider.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.wake.remote.ports;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.reef.tang.annotations.Parameter;
+import org.apache.reef.wake.remote.ports.parameters.TcpPortList;
+import org.apache.reef.wake.remote.ports.parameters.TcpPortListString;
+
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * A TcpPortProvider which gives out ports in a range.
+ */
+public final class ListTcpPortProvider implements TcpPortProvider {
+
+  private static final Logger LOG = Logger.getLogger(ListTcpPortProvider.class.getName());
+  private final List<Integer> tcpPortList;
+
+  /* TODO: Comment out for now due to TANG issue. Must be addressed prior to finalizing pull request.
```

Comment:
>TODO: Comment out for now due to TANG issue. Must be addressed prior to finalizing pull request [](start = 5, length = 95)

Note to self: Resolve this before sending pull request to apache github. This is pending the investigation about Tang injection issue with Lists.

Reply:
Moved over to making the TcpPortList class use sets instead.

---
In reply to: [196509258](https://github.com/sharathmalladi/reef/pull/5#discussion_r196509258) [](ancestors = 196509258)